wagtail/admin/widgets/button.py

```diff
@@ -77,57 +100,73 @@ def __eq__(self, other):
 # Base class for all listing buttons
 # This is also used by SnippetListingButton defined in wagtail.snippets.widgets
 class ListingButton(Button):
-    def __init__(self, label, url, classes=set(), **kwargs):
-        classes = {"button", "button-small", "button-secondary"} | set(classes)
-        super().__init__(label, url, classes=classes, **kwargs)
+    def __init__(self, label="", url=None, classname="", **kwargs):
+        if classname:
+            classname += " button button-small button-secondary"
+        else:
+            classname = "button button-small button-secondary"
+
+        super().__init__(label=label, url=url, classname=classname, **kwargs)
 
 
 class PageListingButton(ListingButton):
-    pass
+    aria_label_format = None
+    url_name = None
+
+    def __init__(
+        self, *args, page=None, next_url=None, attrs={}, page_perms=None, **kwargs
+    ):
+        self.page = page
```

Comment:
Oh I almost forgot, could we perhaps enforce the `page` argument to be required? This class is page-specific (`PageListingButton`) and maybe I missed something but it seems they always get instantiated with a page object. If we enforce this to never be `None`, we can eliminate the `if self.page` in the `url` and other properties, simplifying the logic

Reply:
Enforcing `page` would be a breaking change, as we're telling people to use PageListingButton (with plain label / url params) in the docs for the [register_page_listing_buttons](https://docs.wagtail.org/en/stable/reference/hooks.html#register-page-listing-buttons) hook.

Reply:
Ahh OK, happy to leave it as-is then. I guess I just prefer an early return when possible to prevent adding more indentation than necessary (e.g. in the `url` property), but that's just personal preference